src/k8s/api/v1/cluster_config.go

```diff
@@ -0,0 +1,68 @@
+package v1
+
+type GetClusterConfigRequest struct{}
+
+type GetClusterConfigResponse struct {
+	Config UserFacingClusterConfig
+}
+
+type UpdateClusterConfigRequest struct {
+	Config UserFacingClusterConfig
+}
+
+type UpdateClusterConfigResponse struct {
+}
+
+type UserFacingClusterConfig struct {
```

Comment:
I think we might want to expose stuff other than just functionality configs, the UserFacing is there to improve readability as we already have `ClusterConfig` although still possible since they are in different packages. `types.ClusterConfig` vs `api.ClusterConfig`